modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TenantDeadLetterChannelTestCase.java

```diff
@@ -0,0 +1,208 @@
+                   LogoutAuthenticationExceptionException, XMLStreamException {
+
+        int sendMessageCount = 1;
+
+        String destinationName = "dlctenant1.com/tenantQueue";
+
+        // Create a consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig =
+                new AndesJMSConsumerClientConfiguration("dlctenantuser1!dlctenant1.com", "dlctenantuser1",
+                                                        ExchangeType.QUEUE, destinationName);
+        // Add manual client acknowledgement in configuration
+        consumerConfig
+                .setAcknowledgeMode(JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE);
+        // Acknowledge a message only after 200 messages are received
+        consumerConfig
+                .setAcknowledgeAfterEachMessageCount(200L);
+        consumerConfig.setPrintsPerMessageCount(sendMessageCount);
+
+        // Create consumer client with given consumerConfig
+        AndesClient consumerClient = new AndesClient(consumerConfig, true);
+        // Start consumer client
+        consumerClient.startClient();
+
+        // Create a publisher client configuration
+        AndesJMSPublisherClientConfiguration tenantPublisherConfig =
+                new AndesJMSPublisherClientConfiguration("dlctenantuser1!dlctenant1.com",
+                                                         "dlctenantuser1", ExchangeType.QUEUE, destinationName);
+        tenantPublisherConfig.setNumberOfMessagesToSend(sendMessageCount);
+        tenantPublisherConfig.setPrintsPerMessageCount(sendMessageCount);
+
+        // Create a publisher client with given configuration
+        AndesClient tenantPublisherClient = new AndesClient(tenantPublisherConfig, true);
+        // Start publisher client
+        tenantPublisherClient.startClient();
+
+        AndesClientUtils.waitForMessagesAndShutdown(consumerClient,
+                                                    AndesClientConstants.DEFAULT_RUN_TIME);
+
+        // Get the automation context for the dlctenant1
+        AutomationContext tenantContext = new AutomationContext("MB", "mb001", "dlctenant1", "dlctenantuser1");
+
+        LoginLogoutClient loginLogoutClient = new LoginLogoutClient(tenantContext);
+        String sessionCookie = loginLogoutClient.login();
+        AndesAdminClient andesClient =
+                new AndesAdminClient(super.backendURL, sessionCookie, ConfigurationContextProvider
+                        .getInstance().getConfigurationContext());
+        loginLogoutClient.logout();
+
+
+        Queue tenantUserDlcQueue;
+
+        // Get tenant's dlc queue
+        tenantUserDlcQueue = andesClient.getDlcQueue();
+
+        // Get the automation context for the superTenant
+        AutomationContext superTenantContext = new AutomationContext("MB", "mb001", "superTenant",
+                                                                     "superAdmin");
```

Comment:
Here you can use the following constants instead of "superTenant" and "superAdmin".
org.wso2.carbon.automation.engine.FrameworkConstants.SUPER_TENANT_KEY
org.wso2.carbon.automation.engine.FrameworkConstants.SUPER_TENANT_ADMIN